libraries/eosiolib/contracts/eosio/action.hpp

```diff
@@ -447,8 +447,20 @@ namespace eosio {
     * trans_action.send(st.issuer, to, quantity, memo);
     * @endcode
     */
-   template <eosio::name::raw Name, auto Action>
+   template <eosio::name::raw Name, auto Action, eosio::name::raw DefaultContract = ""_n>
```

Comment:
I don't really like hard-coding an account-name into the action wrapper.  If we're going to do this, I'd rather associate it with the contract class instead.

Reply:
A lot of work is being done by me to clean up this PR.  Should resolve all of the issues mentioned above.